staging/src/k8s.io/apimachinery/pkg/runtime/helper.go

```diff
@@ -257,3 +258,11 @@ func (d WithoutVersionDecoder) Decode(data []byte, defaults *schema.GroupVersion
 	}
 	return obj, gvk, err
 }
+
+
+// FIXME: Probably move it somewhere
+type EncoderWrapper interface {
```

Comment:
Not super excited about this name, maybe SelfEncoder or CustomEncoder?

Reply:
Thanks for suggestions - I'm pretty bad at naming (that's why in KEP I added "better suggestion for name welcome")

Reply:
Changed in KEP.